src/main/java/com/telenordigital/decisionflow/DecisionFlow.java

```diff
@@ -0,0 +1,641 @@
+package com.telenordigital.decisionflow;
+
+import com.telenordigital.decisionflow.Decision.OnAttributesCallback;
+import com.telenordigital.decisionflow.DecisionFlowDescriber.Callback;
+import com.telenordigital.decisionflow.DecisionFlowDescriber.ElementDescriptor;
+import com.telenordigital.decisionflow.DecisionFlowDescriber.ElementType;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.stream.Collectors;
+import org.springframework.expression.Expression;
+import org.springframework.expression.ExpressionParser;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+
+public class DecisionFlow<C, P> implements DecisionMachine<C, P> {
+
+    private InitialNode initialNode = null;
+    private final Map<String, AbstractNode> nodeMap = new HashMap<>();
+
+    private DecisionFlow(final DecisionFlowDescriber describer) {
+        load(describer);
+    }
+
+    public static <C, P> DecisionFlow<C, P> getInstance(final DecisionFlowDescriber describer) {
```

Comment:
Is this class just a wrapper for a "new DecisionFlow<>(describer)" ?
Why not just use "new DecisionFlow<>(describer)" then?